src/deep_links/mod.rs

```diff
@@ -64,6 +64,7 @@ impl From<(&Stream, Option<&Url>, &Settings)> for ExternalPlayerLink {
     ///
     /// [`StreamingServer::base_url`]: crate::models::streaming_server::StreamingServer::base_url
     fn from((stream, streaming_server_url, settings): (&Stream, Option<&Url>, &Settings)) -> Self {
+        let streaming_server_url = streaming_server_url.map(|_| &settings.streaming_server_url);
```

Comment:
```suggestion
        // Only if the streaming server is available, make sure to use the settings url
        // as it might contain a remote url of the server,
        // otherwise, deeplinks will be broken for remote server
        let streaming_server_url = streaming_server_url.map(|_| &settings.streaming_server_url);
```